### PR TITLE
Private Group List Methods

### PR DESCRIPTION
**Pull request by graywolf336**
@RocketChat/core  this enables a `Meteor.call` method to get the list of private groups and also adds a rest api route for getting the list of private groups a user is part of.

This is needed for an external client, a Minecraft plugin, which I am using that needs access to the list of groups as well as channels.

### CODE REVIEW

File: server/methods/groupsList.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+		let options = {
+			fields: { name: 1 },
+			sort: { msgs: -1 }
+		};
+
+		//Verify the limit param is a number
+		if (_.isNumber(limit)) {
+			options.limit = limit;
+		}
+
```
Comment (marceloschmidt) [inline server/methods/groupsList.js]: I think we should also set a default limit and maybe also set a maximum amount of records to fetch. Like `limit = Math.max(options.limit, 50);`

File: server/methods/groupsList.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+			switch (sort) {
+				case 'name':
+					options.sort = { name: 1 };
+					break;
+				case 'msgs':
+					options.sort = { msgs: -1 };
+					break;
+			}
+		}
+
```
Comment (marceloschmidt) [inline server/methods/groupsList.js]: If sort is not set, is usually good to have a default.

**Comment by marceloschmidt** [comment]
Other than the comments above, LGTM

File: server/methods/groupsList.js (hunk 1 above)
Comment (geekgonecrazy) [inline server/methods/groupsList.js]: For sure need a limit.  But also probably need an offset you can specify too

File: server/methods/groupsList.js (hunk 2 above)
Comment (graywolf336) [inline server/methods/groupsList.js]: Default sort is set above a few lines, but it was defaulted to messages and I have changed it to name.

File: server/methods/groupsList.js (hunk 1 above)
Comment (graywolf336) [inline server/methods/groupsList.js]: If we do a limit by default, then paging/skipping needs to also be enabled. I just copied this from the `channelsList` method.

Comment (geekgonecrazy) [inline server/methods/groupsList.js]: Its probably fine to leave out until we get that added.  The list will only include the list the user has.  Which isn't likely to be that large.

Comment (sampaiodiego) [inline server/methods/groupsList.js]: I liked [this](https://github.com/RocketChat/Rocket.Chat/pull/3346/files#diff-c022f65fcc967a0a29880e40b9e80936R55) implementation

Comment (graywolf336) [inline server/methods/groupsList.js]: I think the paging and limiting should happen on the rest api and not the actual method.

**Pull request merged**